### PR TITLE
fix: upgrade jest-diff and jest-matcher-utils

### PR DESCRIPTION
**Pull request by SimenB**
**What**:

Upgrade dated dependencies.

**Why**:

Mostly to dedupe better on new projects, but it also brings in a bugfix (commented inline)

**How**:

`npx npm-check-updates -u`.

**Checklist**:

* [ ] Documentation N/A
* [ ] Tests N/A
* [ ] Updated Type Definitions N/A
* [x] Ready to be merged 
* [ ] Added myself to contributors table

### CODE REVIEW

File: src/to-have-style.js [hunk 1]
```diff
@@ -25,7 +25,6 @@ function printoutStyles(styles) {
     .sort()
     .map(prop => `${prop}: ${styles[prop]};`)
     .join('\n')
-    .concat('\n') // ensure multiline for diff
```
Comment (SimenB) [inline src/to-have-style.js]: Fixed in 23.0.0: https://github.com/facebook/jest/pull/6221

File: package.json [hunk 2]
```diff
@@ -34,13 +34,13 @@
   "dependencies": {
     "chalk": "^2.4.1",
     "css": "^2.2.3",
-    "jest-diff": "^22.4.3",
-    "jest-matcher-utils": "^22.4.3",
-    "pretty-format": "^23.0.1",
+    "jest-diff": "^23.6.0",
+    "jest-matcher-utils": "^23.6.0",
+    "pretty-format": "^23.6.0",
```
Comment (SimenB) [inline package.json]: Not needed as it's within semver range (and `jest-diff` depends on it)m but doesn't hurt to keep them in sync

**Comment by smacpherson64** [comment]
Awesome, Thank you!

**Pull request merged**